api/jsonapi.go

```diff
@@ -1517,8 +1517,10 @@ func (i *jsonAPIHandler) POSTOrderConfirmation(w http.ResponseWriter, r *http.Re
 	}
 
 	// TODO: Remove once broken contracts are migrated
-	if _, err := repo.NewCurrencyCode(contract.BuyerOrder.Payment.Coin); err != nil {
-		log.Warningf("missing contract BuyerOrder.Payment.Coin on order (%s)", conf.OrderID)
+	lookupCoin := contract.BuyerOrder.Payment.Coin
+	_, err = repo.LoadCurrencyDefinitions().Lookup(lookupCoin)
```

Comment:
`NewCurrencyCode` had validation to ensure this value was not empty. Now, instead of only checking that it's valid, we check if it's a known coin by finding out if it's defined by the daemon.